sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -658,4 +659,22 @@ def submit_transaction(self, operations: Iterable[TransactionOperationType], **k
                 "The value of 'operations' must be an iterator "
                 "of Tuples. Please check documentation for correct Tuple format."
             ) from exc
-        return self._batch_send(self.table_name, *batched_requests.requests, **kwargs)  # type: ignore
+
+        try:
+            return self._batch_send(self.table_name, *batched_requests.requests, **kwargs)  # type: ignore
+        except TableTransactionError as ex:
```

Comment:
We should be able to simplify all this code to use the same code path for both Storage and Cosmos - something like this should work:
```python:
except HttpResponseError as error:
    if error.status_code == 400 and not batched_requests.requests:
        return []
    raise
```